### ISSUE TITLE
Not possible to build EasyProg

### ISSUE THREAD

**Issue by JackAsser1337** [KimJorgensen/easyflash#8]
Using the latest version of cc65 you'll get linker error. Last known working git-hash for cc65 is b9c688d9. This will build the .prg but it will crash during start both in VICE and on real hardware.

Which git-hash of cc65 are you using Kim?

**Comment by JackAsser1337** [comment]
Got it to work by using the original crt0.s that comes with cc65. Adding a ONCE segment to the ld2.cfg file will also make it work using the nightly cc65. Want me to submit a PR?

**Comment by KimJorgensen** [comment]
Yes, a PR would be nice so that it possible to use a new version of cc65. Would it still be possible to use an old version of cc65 with your change?

Did you also try to compile EF3BootImage? Hopefully, only one version should be needed to compile all the code in this repository. 

For reference I have been using the tools from https://www.c64-wiki.de/wiki/EasyFlash³/Sourcecode_kompilieren meaning using cc65-2.13.2

**Comment by JackAsser1337** [comment]
It builds now and works in VICE but hangs on real HW. Will need to debug further.

To support both older and newer versions of CC65 the ld65 version test will have to be extended further. I’ll fix that.

Just added support for my new flash chip on my cost reduced EasyFlash1 cartridge I’m developing.

I’ll keep you posted Kim, and Skoe should be CC:ed aswell.

Kind regards, Andreas

21 feb. 2020 kl. 16:56 skrev Kim Jørgensen <notifications@github.com>:

**Comment by KimJorgensen** [comment]
Sounds good.
Did you open source your EasyFlash1 cartridge or announce it anywhere? I would like to take a look at it.
I would have preferred to have my fork merged back into @skoe 's repository but he was quite busy when I asked him 2 years ago. I must admit that haven't asked him since.

**Comment by JackAsser1337** [comment]
I’ll open source it once all bugs are fixed. It’s a 2-chip solution that costs $5 including pcb, components and case to produce.

**Comment by JackAsser1337** [comment]
Almost done now. Sector assignments are a bit more flexible with this chip so care must be taken when erasing some high-sectors. Also write to odd sectors doesn’t work atm. But I’ll get there.

EasyProg hung during timerInitTOD() so I just commented it out for now. Might be my CIA-chip that is blown TBH. I’ve not been exactly nice to this mobo and short ciruit the busses multiple times. :)

Also my cost reduced version doesn’t have any RAM at $dfxx so for now I just places the EAPI ram code at $0100-$017f for all drivers. I’ll fix it later so that only my new driver resides there, or at $0f80- or so.

/Andreas

**Comment by KimJorgensen** [comment]
Oh, so you are developing a new EAPI driver that you want EasyProg to support. Maybe you should keep that in a separate PR if you want that merged into this repository.

If you want your cartridge to be compatible with all EasyFlash releases I think you'll get into trouble for not having any RAM at $dfxx. E.g. all disk2easyflash releases uses that area for storing data and code execution.

**Comment by JackAsser1337** [comment]
Yes, a new EAPI-driver. This is a cost-reduced EF-cart without SRAM. Only
specific titles will support it naturally. I'll make sure EasyProg can
handle the new chip without $dfxx. It's not meant to be fully EF1
compatible. It's meant for game creators to mass produce it easily and
cheap. I'll run my Eye of the Beholder on it.

I'll separate each addition it logical PRs.

Den lör 22 feb. 2020 kl 11:39 skrev Kim Jørgensen <notifications@github.com

**Comment by KimJorgensen** [comment]
Now that I fully understand what you are trying to achieve I don't think that support of a new EF derivative should go into this repository which purpose is to provide Final Cartridge III support. Instead I would suggest that you contact @skoe to get that into the main repository.

I'm still happy to accept a PR that allows people to build the code in this repository with a newer version of cc65, but do understand if you rather would spend you time elsewhere.

It sounds like a nice product that your are developing. Good luck

**Comment by JackAsser1337** [comment]
Yep ok, noted. I’ll provide the PRs once everything has been cleaned.

/A
